pkg/fargate/coredns/coredns_test.go

```diff
@@ -165,6 +180,23 @@ var _ = Describe("coredns", func() {
 				Expect(err.Error()).To(Equal("timed out while waiting for \"coredns\" to be scheduled on Fargate"))
 			}
 		})
+
+		It("Should timeout if coredns pods do not have the correct annotations", func() {
+			failureCases := [][]runtime.Object{
+				{deploymentWithAnnotations("fargate", 0, 2, nil), podWithAnnotations("fargate", v1.PodPending, nil), podWithAnnotations("fargate", v1.PodRunning, nil)},
+				{deploymentWithAnnotations("ec2", 0, 2, nil), podWithAnnotations("ec2", v1.PodFailed, nil), podWithAnnotations("ec2", v1.PodRunning, nil)},
+				{deploymentWithAnnotations("ec2", 0, 2, map[string]string{}), podWithAnnotations("ec2", v1.PodRunning, map[string]string{}), podWithAnnotations("ec2", v1.PodRunning, map[string]string{})},
+			}
+			for _, failureCase := range failureCases {
+				// Given:
+				mockClientset := mockClientsetWith(failureCase...)
+				// When:
+				err := coredns.WaitForScheduleOnFargate(mockClientset, retryPolicy)
+				// Then:
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("timed out while waiting for \"coredns\" to be scheduled on Fargate"))
```

Comment:
This can be simplified to

```suggestion 
				Expect(err).To(MatchError("timed out while waiting for \"coredns\" to be scheduled on Fargate"))
```